pyvista/core/composite.py

```diff
@@ -690,3 +703,244 @@ def copy(self, deep=True):
+            points or cells, it will prioritize an array matching this
+            type.  Can be either ``'cell'`` or ``'point'``.
+
+        allow_missing : bool, optional
+            Allow missing scalars in part of the composite dataset. If all
+            blocks are missing the array, it will raise a ``KeyError``.
+
+        Returns
+        -------
+        pyvista.FieldAssociation
+            Field association of the scalars activated.
+
+        numpy.ndarray
+            An array from the dataset matching ``name``.
+
+        Notes
+        -----
+        The number of components of the data must match.
+
+        """
+        data_assoc: List[Tuple[FieldAssociation, np.ndarray, MultiBlock]] = []
+        for block in self:
+            if block is not None:
+                if isinstance(block, MultiBlock):
+                    field, scalars = block.set_active_scalars(
+                        name, preference, allow_missing=allow_missing
+                    )
+                else:
+                    try:
+                        field, scalars = block.set_active_scalars(name, preference)
+                    except KeyError as err:
+                        if not allow_missing:
+                            raise err
+                        block.set_active_scalars(None, preference)
+                        field, scalars = FieldAssociation.NONE, np.array([])
+
+                if field != FieldAssociation.NONE:
+                    data_assoc.append((field, scalars, block))
+
+        if name is None:
+            return FieldAssociation.NONE, np.array([])
+
+        if not data_assoc:
+            raise KeyError(f'"{name}" is missing from all the blocks of this composite dataset.')
+
+        field_asc = data_assoc[0][0]
+        # set the field association to the preference if at least one occurrence
+        # of it exists
+        if field_asc.name.lower() != preference:
+            for field, _, _ in data_assoc:
+                if field.name.lower() == preference:
+                    field_asc = getattr(FieldAssociation, preference.upper())
+                    break
+
+        # Verify array consistency
+        dims: Set[int] = set()
+        dtypes: Set[np.dtype] = set()
+        for block in self:
+            for field, scalars, _ in data_assoc:
+                # only check for the active field association
+                if field != field_asc:
+                    continue
+                dims.add(scalars.ndim)
+                dtypes.add(scalars.dtype)
+
+        if len(dims) > 1:
+            raise ValueError(f'Inconsistent dimensions {dims} in active scalars.')
+
+        # check complex mismatch
+        is_complex = [np.issubdtype(dtype, np.complexfloating) for dtype in dtypes]
+        if any(is_complex) and not all(is_complex):
+            raise ValueError('Inconsistent complex and real data types in active scalars.')
+
+        return field_asc, scalars
+
+    def as_polydata(self):
```

Comment:
The name is slightly confusing, but I guess the docstring is clear enough, and I don't have a better name in mind. Perhaps something `as_polydata_blocks()` or better would make it clear that the encapsulating type won't change, only the contents. Mild suggestion.

Reply:
Agreed. `as_polydata` sounds like we're just casting or converting to `pyvista.PolyData`. Fixed this in aee6ffa58ee0f26d183b464f761ef056f110cae2.